contracts/mocks/MockOracle.sol

```diff
@@ -65,7 +75,15 @@ contract MockOracle {
     }
 
     function getExpiryPrice(address _asset, uint256 _expiryTimestamp) external view returns (uint256, bool) {
-        return (storedPrice[_asset][_expiryTimestamp], isFinalized[_asset][_expiryTimestamp]);
+        uint256 price = stablePrice[_asset];
+        bool _isFinalized = true;
+
+        if (price == 0) {
+            price = storedPrice[_asset][_expiryTimestamp];
+            _isFinalized = isFinalized[_asset][_expiryTimestamp];
```

Comment:
isn't using isFinalized from the storage (`isFinalized[_asset][_expiryTimestamp]; `) makes it something that need to be poked on?